glacier/src/reducers/fields.ts

```diff
@@ -0,0 +1,23 @@
+import {FieldState, Field} from "../model";
+import {AllActions} from "../actions";
+
+export function fields(state: FieldState | undefined, action: AllActions) {
+     if (!state) return {};
+     if (action.error) {
+         throw action.error;
+     }
+     switch (action.type) {
+         case "ADD_FIELD": {
+            return {...state, ...action.payload};
+         }
+         case "REMOVE_FIELD": {
+             let filteredState: Field[] = [];
+             for (const field of action.payload) {
+                 filteredState = state.fields.filter(item =>
+                     field.name !== item.name && field.table !== item.table);
+             }
+             return {filteredState};
```

Comment:
This should literally just be `return filteredState`. `return {filteredState}` will create an object like so: `{filteredState: [{a}, {b}, {c}]}`.